### PR TITLE
Fix #1362 store exceptions thrown in port forwarder websocket

### LINKED ISSUE #1362: Pod Port Forward Error Handling
Currently, the only indicator that a port forwarder websocket was closed due to an exception is the log output.
Instead of mere logging these exceptions, I'd like to store them, so callers may implement their own error handling. Callers need to be able to (1) check whether there was an error that caused the websocket to close and (2) get a list of exceptions that were thrown.

### PR DESCRIPTION
**Pull request by dotstone**
Fix #1362

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,8 @@
     * Fix #1348: support `v1beta1` version for `ExecCredentials`
 
     * Fix #1354: suppress log warnings that `CustomResourceDefinition`s are still in beta
+    
+    * Fix #1362: store exceptions thrown in port forwarder websocket
```
Comment (rohanKanojia) [inline CHANGELOG.md]: 4.1.2 was released a few days ago, Could you please add this to a new section `4.1-SNAPSHOT`?

Comment (dotstone) [reply CHANGELOG.md]: Thanks. Done

**Comment by dotstone** [comment]
After adapting the changelog.md, the build failed due to:
```
Feb 19, 2019 11:25:00 AM org.arquillian.cube.kubernetes.impl.resources.KubernetesResourcesApplier removeKubernetesResourcesAtClassScope
INFO: Deleting environment for io.fabric8.kubernetes.PodIT
[ERROR] Tests run: 7, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 13.193 s <<< FAILURE! - in io.fabric8.kubernetes.PodIT
[ERROR] update(io.fabric8.kubernetes.PodIT)  Time elapsed: 0.342 s  <<< ERROR!
io.fabric8.kubernetes.client.KubernetesClientException: 
Failure executing: PATCH at: https://127.0.0.1:8443/api/v1/namespaces/itest-108ce97b/pods/pod1-meng1l. Message: Pod "pod1-meng1l" is invalid: spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or `spec.tolerations` (only additions to existing tolerations)
```
Do you have an idea about what went wrong there?

**Comment by rohanKanojia** [comment]
@dotstone : it's not you. It's us. Our CI gets overloaded sometimes and misbehaves ;-)

**Comment by dotstone** [comment]
I updated the PR to split client from server exceptions so callers know whether the problem is originating from the k8s cluster or from the client.

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**